transport/src/main/java/io/netty/bootstrap/Bootstrap.java

```diff
@@ -189,7 +189,13 @@ private ChannelFuture doResolveAndConnect0(final Channel channel, SocketAddress
                                                final SocketAddress localAddress, final ChannelPromise promise) {
         try {
             final EventLoop eventLoop = channel.eventLoop();
-            final AddressResolver<SocketAddress> resolver = this.resolver.getResolver(eventLoop);
+            AddressResolver<SocketAddress> resolver;
+            try {
+                resolver = this.resolver.getResolver(eventLoop);
+            } catch (Throwable e) {
+                channel.close();
```

Comment:
@blindpirate I think this should be changed to:

```
try {
    ...
} catch (Throwable cause) {
    channel.close();
    promise.setFailure(e);
    return;
}

```

Reply:
Also would be nice if you could write a unit test for it and include it in `BootstrapTest`.

Reply:
OK, will try.

Reply:
@blindpirate please ping me once done ...

Reply:
actually let me just take care and add me as an co-author then.